src/tests/roc_audio/test_mixer.cpp

```diff
@@ -20,11 +20,11 @@ namespace audio {
 
 namespace {
 
-enum { BufSz = 100, MaxSz = 500 };
+enum { BufSz = 100, FrameLength = (core::nanoseconds_t) 7000000, SampleRate = 44100, ChannelMask = 0x1 };
 
 core::HeapAllocator allocator;
-core::BufferPool<sample_t> buffer_pool(allocator, MaxSz, true);
-core::BufferPool<sample_t> large_buffer_pool(allocator, MaxSz * 10, true);
+core::BufferPool<sample_t> buffer_pool(allocator, FrameLength, true);
+core::BufferPool<sample_t> large_buffer_pool(allocator, FrameLength * 10, true);
```

Comment:
Now we're passing FrameLength both to Mixer and BufferPool. This is also wrong: BufferPool accepts size in bytes, and Mixer accepts duration in nanoseconds.

I guess we should pass FrameLength to Mixer and pass FrameLength converted to bytes to BufferPool.

Reply:
We should also fix other tests. FrameLength and sizes passed to pools should correspond to each other.

Reply:
I replaced 'MaxSz' with 'FrameLength'. Should I replace 'BufSz' with it's corresponding frame length value and convert it to bytes before using it?

Reply:
Look, you can't just replace MaxSz with FrameLength.

Before this PR, we used number of samples everywhere. Now we're using number of samples in some places and number of nanoseconds in other places.

In particular:

* Pool constructor accepts number of samples, as before
* Mixer constructor now accepts number of nanoseconds
* reader.add() accepts number of samples, as before
* expect_output() accepts number of samples, as before

In these tests I'd suggest to use three constants:

* `BufSz` - "regular" buffer size, number of samples
* `MaxBufSz` - maximum buffer size, number of samples
* `MaxBufDuration` - duration corresponding to the maximum buffer size, number of nanoseconds

And use them as follows:

* pass MaxBufDuration to Mixer constructor
* pass MaxBufSz to Pool constructor (this part remains unchanged)
* pass either BufSz or MaxBufSz to reader.add() and expect_output(), depending on the test (this part remains unchanged)

We can define MaxBufDuration to be calculated from MaxBufSz, or vice versa. I'd suggest to leave MaxBufSz equal to 500, as before, and to calculate MaxBufDuration from it.

Reply:
And I think we should use the same approach in other tests as well. I.e. define maximum size and calculate corresponding duration from it.

Reply:
And it'd better to use some similar names, e.g. `MaxBufSize` + `MaxBufDuration`, as suggested above, to make it clear that these two constants are related.

Reply:
done.